normalizer_lib.py

```diff
@@ -1,205 +1,224 @@
 #/usr/bin/env python
 """Library of language-agnostic FST rewrite rules to normalize text."""
 
+import importlib
 import unicodedata
-from dataclasses import dataclass
 from typing import List
 from pynini import *
 from pynini.lib import byte, pynutil
-import config.zu as zu
 
-LANGUAGE = zu
+class NormalizerLib:
+    """Loads language-specific information for the text normalizer."""
+
+    def __init__(self, language):
+        self.language = importlib.import_module("config."+language)
+        self.graphemes = self.language.GRAPHEMES
+        self.initial_punctuation = self.language.INITIAL_PUNCTUATION
+        self.final_punctuation = self.language.FINAL_PUNCTUATION
+        self.other_punctuation = union(r"\[", r"\]")
+        self.punctuation = union(self.initial_punctuation,
+                                 self.final_punctuation,
+                                 self.other_punctuation)
+        self.numerals = self.language.NUMERALS
+        try:
+            self.language_specific_preprocessing = self.language.LANGUAGE_SPECIFIC_PREPROCESSING
+        except Exception:
+            self.language_specific_preprocessing = None
+
+    SIGMA_STAR = byte.BYTES.closure()
+    UNDERSCORE = acceptor("_")
+    DUMMY_REWRITE = cdrewrite(cross("", ""), "", "", SIGMA_STAR)
+
+    def verbalizable(self):
 
-GRAPHEMES = LANGUAGE.GRAPHEMES
-
-INITIAL_PUNCTUATION = LANGUAGE.INITIAL_PUNCTUATION
-FINAL_PUNCTUATION = LANGUAGE.FINAL_PUNCTUATION
-OTHER_PUNCTUATION = union(r"\[", r"\]")
-PUNCTUATION = union(INITIAL_PUNCTUATION, FINAL_PUNCTUATION, OTHER_PUNCTUATION)
-NUMERALS = LANGUAGE.NUMERALS
-UNDERSCORE = acceptor("_")
-SIGMA_STAR = byte.BYTES.closure()
-
-
-@dataclass
-class Verbalizable:
-    """Contains verbalizable tokens."""
-
-    @staticmethod
-    def verbalizable():
         """Returns union of verbalizable tokens.
 
-        args: none
+        Args: None
 
-        returns: Union of acceptors for verbalizable tokens.
+        Returns: Union of acceptors for verbalizable tokens.
         """
-        token = (INITIAL_PUNCTUATION.ques +
-                 (GRAPHEMES.plus | NUMERALS.plus).plus +
-                 FINAL_PUNCTUATION.ques)
+        token = (self.initial_punctuation.ques +
+                 (self.graphemes.plus | self.numerals.plus).plus +
```

Comment:
Are these .plus inside the statement redundant?

Reply:
I believe they are.